internal/services/sentinel/sentinel_alert_rule_fusion_resource.go

```diff
@@ -158,25 +162,69 @@ func resourceSentinelAlertRuleFusionCreateUpdate(d *pluginsdk.ResourceData, meta
 		},
 	}
 
-	if !d.IsNewResource() {
-		resp, err := client.Get(ctx, id)
-		if err != nil {
-			return fmt.Errorf("retrieving %q: %+v", id, err)
+	if _, err := client.CreateOrUpdate(ctx, id, params); err != nil {
+		return fmt.Errorf("creating Sentinel Alert Rule Fusion %q: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceSentinelAlertRuleFusionRead(d, meta)
+}
+
+func resourceSentinelAlertRuleFusionUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.AlertRulesClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := alertrules.ParseAlertRuleID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving %q: %+v", id, err)
+	}
+
+	if err = assertAlertRuleKind(resp.Model, alertrules.AlertRuleKindFusion); err != nil {
+		return fmt.Errorf("asserting alert rule of %q: %+v", id, err)
+	}
+
+	payload := resp.Model.(alertrules.FusionAlertRule)
+
```

Comment:
This doesn't follow the pattern that we expect for the update function which
1. nil check Model
2. nil check Properties
3. Patch changes in to Properties/Model

This particular case will deviate slightly since this is a discriminated type and it's generally good practice to check that the type assertion we're doing is successful and throwing an error if it isnt. Can you please update this to 
```suggestion
	if resp.Model == nil {
		return fmt.Errorf("retrieving %s: `model` was nil", id)
	}
	
	payload, ok := resp.Model.(alertrules.FusionAlertRule)
	if !ok {
		return fmt.Errorf("retrieving %s: expected an alert rule of type `Fusion`, got %q", pointer.From(resp.Model.AlertRule().Type))
	}
	
	if payload.Properties == nil {
		return fmt.Errorf("retrieving %s: `properties` was nil", id)
	}
```